### ISSUE TITLE
Future improvements: Strategies to increase model accuracy

### ISSUE THREAD

**Issue by aquaticcalf** [fishcareyolo/fishcareyolo#5]
## Strategies to Improve Model Accuracy

This issue documents approaches to improve detection accuracy for future iterations.

### 1. More Training Data (Most Effective)
- Collect more labeled fish images, especially for underrepresented classes
- Use data augmentation tools to generate synthetic variations
- Consider sourcing from multiple fish farms/environments for diversity

### 2. Better Data Quality
- Audit existing labels for accuracy
- Remove duplicate or near-duplicate images
- Ensure consistent labeling standards across annotators
- Balance class distribution (avoid having too many "healthy" vs disease samples)

### 3. Larger Model Architecture
```bash
# Current: YOLOv8n (nano) - optimized for mobile
# Try: YOLOv8s (small) or YOLOv8m (medium) for higher accuracy

# Edit mina/train.py or pass different pretrained weights
uv run mina-train --epochs 100 --batch 16  # uses yolov8n.pt by default
```

| Model | Size | mAP | Speed | Use Case |
|-------|------|-----|-------|----------|
| YOLOv8n | 3.2MB | Lower | Fastest | Mobile (current) |
| YOLOv8s | 11.2MB | Medium | Fast | Mobile/Edge |
| YOLOv8m | 25.9MB | Higher | Moderate | Server-side |
| YOLOv8l | 43.7MB | Higher | Slower | Server-side |

Trade-off: Larger models won't run efficiently on mobile devices.

### 4. Hyperparameter Tuning
- Learning rate: Try `lr0=0.001` or `lr0=0.02`
- Batch size: Larger batches (32, 64) if GPU memory allows
- Image size: Try `--imgsz 1280` for more detail (slower)
- Augmentation: Adjust `mosaic`, `mixup`, `hsv_*` parameters

### 5. Training Techniques
- **Longer training**: Increase epochs (with early stopping to prevent overfitting)
- **Learning rate scheduling**: Already uses cosine annealing
- **Multi-scale training**: Train on varying image sizes
- **Test-time augmentation (TTA)**: Ensemble predictions from augmented inputs

### 6. Ensemble Methods
- Train multiple models with different seeds
- Average predictions from YOLOv8n + YOLOv8s
- Increases accuracy but also inference time

### Priority Recommendation

For maximum impact with minimum effort:
1. **More diverse training data** - biggest gains
2. **Fix any labeling errors** - free accuracy boost
3. **Try YOLOv8s** - if mobile size constraints allow

---

*Note: Diminishing returns apply. Going from 90%→95% is much harder than 80%→90%.*
